agent-common/src/com/thoughtworks/go/agent/launcher/ServerCall.java

```diff
@@ -45,6 +45,11 @@ public static ServerResponseWrapper invoke(HttpMethod method) throws Exception {
         HashMap<String, String> headers = new HashMap<String, String>();
         HttpClient httpClient = new HttpClient();
         httpClient.setConnectionTimeout(HTTP_TIMEOUT_IN_MILLISECONDS);
+        final String proxyHost = System.getProperty("http.proxyHost");
+        final String proxyPort = System.getProperty("http.proxyPort");
+        final String nonProxyHosts = System.getProperty("http.nonProxyHosts");
```

Comment:
Would you consider adding support for `https.proxyHost` and `https.proxyPort` as well?